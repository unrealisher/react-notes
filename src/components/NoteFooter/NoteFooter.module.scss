.footer {
  margin-top: auto;
  display: flex;
  flex-direction: column;
}

.wrapper {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
