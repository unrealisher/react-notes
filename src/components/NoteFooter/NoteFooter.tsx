import React from "react";

import Tags from "./../Tags/Tags";
import Edit from "./../Edit/Edit";
import Date from "./../Date/Date";

import styles from "./NoteFooter.module.scss";

interface IProps {
  tags?: string[];
  created?: number;
}

const NoteFooter = (props: IProps): JSX.Element => {
  const { tags, created }: IProps = props;
  return (
    <div className={styles.footer}>
      {tags !== undefined ? <Tags tags={tags} /> : null}
      <div className={styles.wrapper}>
        <Edit />
        <Date created={created} />
      </div>
    </div>
  );
};

export default NoteFooter;
