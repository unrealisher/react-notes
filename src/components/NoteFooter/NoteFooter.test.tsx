import React from "react";
import { shallow } from "enzyme";
import { NoteFooter } from "./NoteFooter";

describe("NoteFooter", () => {
  const activeNotes = true;
  const note = {
    id: 1,
    type: "text",
    title: "Завтра дедлайн",
    text: "Не забыть сверстать макет",
    tags: [1, 2],
    color: 2,
    size: "s",
    created: 1551714600000
  };
  const tags = ["Покупки", "Работа", "ШРИ"];
  const setPatchItem = (): void => {};
  const setPopup = (): void => {};
  it("correct render", () => {
    const component = shallow(
      <NoteFooter
        activeNotes={activeNotes}
        note={note}
        tags={tags}
        setPatchItem={setPatchItem}
        setPopup={setPopup}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
