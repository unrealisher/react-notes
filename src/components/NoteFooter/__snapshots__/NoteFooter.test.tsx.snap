// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteFooter correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteFooter
    activeNotes={true}
    note={
      Object {
        "color": 2,
        "created": 1551714600000,
        "id": 1,
        "size": "s",
        "tags": Array [
          1,
          2,
        ],
        "text": "Не забыть сверстать макет",
        "title": "Завтра дедлайн",
        "type": "text",
      }
    }
    setPatchItem={[Function]}
    setPopup={[Function]}
    tags={
      Array [
        "Покупки",
        "Работа",
        "ШРИ",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Tags
          tags={
            Array [
              "Покупки",
              "Работа",
              "ШРИ",
            ]
          }
        />,
        <div
          className="wrapper"
        >
          <Connect(Edit)
            note={
              Object {
                "color": 2,
                "created": 1551714600000,
                "id": 1,
                "size": "s",
                "tags": Array [
                  1,
                  2,
                ],
                "text": "Не забыть сверстать макет",
                "title": "Завтра дедлайн",
                "type": "text",
              }
            }
            setPatchItem={[Function]}
            setPopup={[Function]}
          />
          <Date
            created={1551714600000}
          />
        </div>,
      ],
      "className": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "tags": Array [
            "Покупки",
            "Работа",
            "ШРИ",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Edit)
              note={
                Object {
                  "color": 2,
                  "created": 1551714600000,
                  "id": 1,
                  "size": "s",
                  "tags": Array [
                    1,
                    2,
                  ],
                  "text": "Не забыть сверстать макет",
                  "title": "Завтра дедлайн",
                  "type": "text",
                }
              }
              setPatchItem={[Function]}
              setPopup={[Function]}
            />,
            <Date
              created={1551714600000}
            />,
          ],
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "note": Object {
                "color": 2,
                "created": 1551714600000,
                "id": 1,
                "size": "s",
                "tags": Array [
                  1,
                  2,
                ],
                "text": "Не забыть сверстать макет",
                "title": "Завтра дедлайн",
                "type": "text",
              },
              "setPatchItem": [Function],
              "setPopup": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "created": 1551714600000,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Tags
            tags={
              Array [
                "Покупки",
                "Работа",
                "ШРИ",
              ]
            }
          />,
          <div
            className="wrapper"
          >
            <Connect(Edit)
              note={
                Object {
                  "color": 2,
                  "created": 1551714600000,
                  "id": 1,
                  "size": "s",
                  "tags": Array [
                    1,
                    2,
                  ],
                  "text": "Не забыть сверстать макет",
                  "title": "Завтра дедлайн",
                  "type": "text",
                }
              }
              setPatchItem={[Function]}
              setPopup={[Function]}
            />
            <Date
              created={1551714600000}
            />
          </div>,
        ],
        "className": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "tags": Array [
              "Покупки",
              "Работа",
              "ШРИ",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Edit)
                note={
                  Object {
                    "color": 2,
                    "created": 1551714600000,
                    "id": 1,
                    "size": "s",
                    "tags": Array [
                      1,
                      2,
                    ],
                    "text": "Не забыть сверстать макет",
                    "title": "Завтра дедлайн",
                    "type": "text",
                  }
                }
                setPatchItem={[Function]}
                setPopup={[Function]}
              />,
              <Date
                created={1551714600000}
              />,
            ],
            "className": "wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "note": Object {
                  "color": 2,
                  "created": 1551714600000,
                  "id": 1,
                  "size": "s",
                  "tags": Array [
                    1,
                    2,
                  ],
                  "text": "Не забыть сверстать макет",
                  "title": "Завтра дедлайн",
                  "type": "text",
                },
                "setPatchItem": [Function],
                "setPopup": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "created": 1551714600000,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
